bson/bsoncodec/default_value_decoders.go

```diff
@@ -62,15 +63,88 @@ func (dvd DefaultValueDecoders) RegisterDefaultDecoders(rb *RegistryBuilder) {
 		RegisterDefaultDecoder(reflect.Struct, &StructCodec{cache: make(map[reflect.Type]*structDescription), parser: DefaultStructTagParser})
 }
 
+// getValue check if value is a pointer and return final value to
+func getValue(val reflect.Value) reflect.Value {
+	if val.Kind() == reflect.Ptr {
+		if val.IsNil() {
+			val.Set(reflect.New(val.Type().Elem()))
+		}
+		return val.Elem()
+	}
+	return val
+}
+
+// getValueKind get value's kind or it's element's kind
+func getValueKind(val reflect.Value) reflect.Kind {
+	if val.Kind() == reflect.Ptr {
+		return val.Type().Elem().Kind()
+	}
+	return val.Kind()
+}
+
+// checkTypeError check ValueReader is one of required bsontypes, an error is returned otherwise
+func checkTypeError(vr bsonrw.ValueReader, requiredType ...bsontype.Type) error {
+	currentType := vr.Type()
+	for _, t := range requiredType {
+		if currentType == t {
+			return nil
+		}
+	}
+	typeString := []string{}
+	for _, t := range requiredType {
+		var payload string
+		switch t {
+		case bsontype.Decimal128:
+			payload = "decimal.Decimal128"
+		default:
+			payload = t.String()
+		}
+		typeString = append(typeString, payload)
+	}
+	return fmt.Errorf("cannot decode %v into a %s", vr.Type(), strings.Join(typeString, ", "))
+}
+
 // BooleanDecodeValue is the ValueDecoderFunc for bool types.
 func (dvd DefaultValueDecoders) BooleanDecodeValue(dctx DecodeContext, vr bsonrw.ValueReader, i interface{}) error {
-	if vr.Type() != bsontype.Boolean {
-		return fmt.Errorf("cannot decode %v into a boolean", vr.Type())
+	var (
```

Comment:
Instead of using a multiline `var` block, can you please change this to individual `var` statements?

Reply:
all var block in default_value_decoders is changed